app/assets/stylesheets/darkswarm/modals.css.scss

```diff
@@ -28,7 +28,7 @@ dialog, .reveal-modal {
   // Medium and up - when modal IS NOT full screen
   @media only screen and (min-width: 641px) {
     top: 10%;
-    max-height: 120%;
```

Comment:
on my almost-merged cookies work I have changed this max-height to 100%. On the cookies policy page modal I want it to go to until the end of the screen: lots of content. Maybe we can set this to 95% to cover both cases?

Reply:
@luisramos0 Thanks for the heads-up! Seeing that your cookies banner has been merged into by now. I will rebase this branch upon master.

Hmm... If we retain the 10% position at the top but use a max height of 95%, we might end up with some hidden content again on some screen dimensions.

I see that your changes apply a CSS class "cookies-policy-modal" to the modal. I could try just adding custom styles that sets "top" to 2% and "max-height" to 96% (and animation to start at -2%) only for this particular modal. It would render similar to how it does in master now.

Are you okay with this?

Reply:
yes, cookies policy modals can take it. I was just trying to avoid the !important:
  top: 2% !important;
  max-height: 96% !important;

Do you know how can we avoid these !importants?
One easy trick I used in the past is to have the generic/common css files (like modals.css in this case) be on the top of the all.css file. I am not sure how the stylesheets are loaded into all.css but it may well be alphabetically and because Modals comes after Cookies the styles on modals are read after and get the priority :-(

Any suggestions?

Reply:
Ah! Yes, you're right, the files would be loaded in alphabetical order, so `cookies_policy_modal.css.scss` would be loaded before `modals.scss`.

I'm seeing two options:

1. Use more specific selectors for the cookies policy modal, like this:

    ```css
    .cookies-policy-modal.reveal-modal { }
    ```

2. Load the CSS partial for the cookies policy modal last.

I think the second is more appropriate.

In my experience, page-specific CSS (I would categorize cookies policy modal CSS under this) are best to be loaded last, and could be loaded in any order among themselves because they are not reused.

So far, the file structure under `darkswarm/` is flat. I've been a fan of the [7-1 Pattern](http://sass-guidelin.es/#the-7-1-pattern), which guides the file structure of CSS partials and helps make their load order easier to manage. Following this, I would move the CSS specific to the cookies policy modal under `pages/`.

The following would then make sure that all the other CSS would still load in the way that they already do:

```scss
@import '*'; // This will not import subdirectories.
@import 'pages/*';
```

For a start, I would also move CSS for the other modals under `pages/`. The rest of the files, we could reorganize as we go along, or when we could properly audit. I'm also not sure yet how OFN instances make CSS customizations, so not sure how messy that would be.

What do you think?

Reply:
perfect! 7.1 is fine. i am a fan of funcional organizations so id put cookiespolicy css next to the js files and htnl of that page. but rails is always organizing things by type(controllers/models/etc), which I dont like.
Anyway, what you described sounds awesome and if support that change!
i dont think instances have css customizations.
thanks!

Reply:
I found that the modal library does not allow us to easily customize the top position of just certain modals either. The faux modal it inserts to the DOM, from which it determines the top position of the real modal, does not apply any custom modal CSS class we use for the real modal.

To work around this limitation, I changed modals for large screens from 10% position from top and 80% max height, to 5% position from top and 90% max height. Before the recent changes, most modals were rendering at 10px top position (not the originally planned 10%), so changing this to 5% doesn't seem to be much of a compromise.

Reply:
I split `**/*` into a couple of lines, to allow us to start specifying the order in which we import the CSS partials. The files in `pages/*` are loaded last.

About the taller cookies policy modal, it seems that it is not easy to apply custom position and height to modals - this doesn't seem to be supported by the modal library.

To work around this limitation, I went ahead and made a general change for all modals in large screens to start from 5% and have 90% max height, instead of 10% and 80%. Before these changes, most modals were rendering at 10px top position, not the originally planned 10%. Changing this to 5% doesn't seem to be much of a compromise, and it works out nicely.

Let me know what you think.